python/graphstorm/eval/evaluator.py

```diff
@@ -830,6 +830,298 @@ def compute_score(self, pred, labels, train=True):
 
         return scores
 
+class GSgnnLPEvaluator(GSgnnBaseEvaluator, GSgnnLPRankingEvalInterface):
```

Comment:
Would it be possible to have a single class for LP evaluation that is a super-set of the individual MRR and Hits@ classes? 

Seems like their constructors are very similar so we could have a single class that performs all the calculations instead of 3? Thinking about future maintenance otherwise we'd add a whole new class for each new metric?

Reply:
This is right the super-set of individual metric classes, and can be used for all new metrics as well. I just left the previous MRR and Hits evaluators there.

If you all agree, I can remove those single metric evaluators.

Reply:
Do you mean have a single class for GSgnnLPEvaluator and GSgnnPerEtypeLPEvaluator?

Reply:
I think GSgnnLPEvaluator is going to replace the GSgnnMrrLPEvaluator and GSgnnHitsLPEvaluator.

Can we add a deprecation note for GSgnnMrrLPEvaluator and GSgnnHitsLPEvaluator?

Reply:
> Do you mean have a single class for GSgnnLPEvaluator and GSgnnPerEtypeLPEvaluator?

If possible yes, their interface seems to be the same, the only thing that changes is the set of metrics they calculate?

Reply:
Yes. I will add deprecation note for MRR and Hits evaluators.